### ISSUE TITLE
Prettier/WP-Prettier and potential consistency issue with legacy-peer-deps

### ISSUE THREAD

**Issue by martinDolan** [WebDevStudios/wds-bt#12]
**Recommendation Options:**

1. Consider setting as a dev-dependency

2. Consider adding `.npmrc` to the project with `legacy-peer-deps=true` so that `npm install` will consistently install wp-prettier.

3. Update documentation to let people know that if their global `.npmrc` file has `legacy-peer-deps=true`, they will probably have a different linter.
----
**More detailed explanation:**

I was struggling with an issue where my version of Prettier was not in parity with others from our team. Eventually I realized (thanks for the 👀 @Cheffheid) I had `legacy-peer-deps=true` in my `.npmrc` in my personal user's folder. `/Users/moconnor/.npmrc`. 

Having this set to true causes `"name": "wp-prettier",` to be set in in `node_modules/prettier/package.json`. Without `legacy-peer-deps=true` it sets it as `"name": "prettier",`

This caused my linter to see errors that others didn't see, like this:
```
ERROR in [eslint] 
/Users/martinoconnor/Local Sites/wds-bt-testing-2/app/public/wp-content/themes/wds-bt/assets/js/global/header.js
   1:27  error  Insert `·`                                                              prettier/prettier
   2:30  error  Replace `'js-enabled'` with `·'js-enabled'·`                            prettier/prettier
   3:2   error  Insert `·`                                                              prettier/prettier
   5:43  error  Replace `'.site-header'` with `·'.site-header'·`                        prettier/prettier
  11:6   error  Replace `siteHeader` with `·siteHeader·`                                prettier/prettier
  15:7   error  Replace `scrollTop·>·lastScrollTop` with `·scrollTop·>·lastScrollTop·`  prettier/prettier
  26:6   error  Replace `!isThrottled` with `·!·isThrottled·`                           prettier/prettier
  30:14  error  Insert `·`                                                              prettier/prettier
  32:22  error  Insert `·`                                                              prettier/prettier
  36:25  error  Replace `'scroll',·throttleScroll` with `·'scroll',·throttleScroll·`    prettier/prettier
  ```

However, I think this is actually correct for `WP Prettier`. If you look at line 14 of `node_modules/@wordpress/prettier-config/lib/index.js` you'll see:
`const isWPPrettier = prettierPackage.name === 'wp-prettier';` which is looking for the package to be called `wp-prettier` in order to receive all of the correct wp-prettier linting overrides.

Adding the `legacy-peer-deps=true` to an `.npmrc` will get the same version which is named `wp-prettier`.

**Comment by khleomix** [comment]
@martinDolan Will take a look at this. There is actually two packages available, the main `prettier` one and the `wp-prettier` which was forked by Automattic https://github.com/Automattic/wp-prettier and is currently way behind `prettier`. I'll test it out though and see if this is something we should use instead.

**Comment by Cheffheid** [comment]
I think the fact that there are two packages is part of the problem we encountered.

Running `npm i` without the `--legacy-peer-deps` flag will get you the regular ol' `prettier` package at the end of the install process, but if you run npm install with that flag enabled (and Marty happened to have it enabled globally through his config file, which is how we found this out) you will end up with `wp-prettier` and a bunch of files that will fail the linter checks out of the box.

It _looks_ like wp-prettier is coming along as a dependency for [wp-scripts](https://github.com/WordPress/gutenberg/blob/ab491433332fe96158fe47f920ce099c64656842/packages/scripts/package.json#L77) in that latter case.

And so I think part of the decision here needs to be whether or not we want prettier to enforce the spaces inside of parentheses or ignore that part of the WordPress code standards, because that is the main thing that wp-prettier does and it's what [WordPress likes to do](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#spacing) (and what Prettier doesn't seem to want to support).
